docs/userGuide/syntax/lists.md

```diff
@@ -109,3 +109,127 @@ first number
+
+**To customize list icons, add the configuration `{icon="icon-name"}` after a specific list item**
+
+<include src="codeAndOutput.md" boilerplate >
+<variable name="highlightStyle">markdown</variable>
+<variable name="code">
+* Item 1 {icon="glyphicon-education"}
+* Item 2
+  * Item 2.1 {icon="fas-file-code"}
+  * Item 2.2
+* Item 3 {icon="fas-code-branch"}
+  * Item 3.1 
+* Item 4 {icon="octicon-git-pull-request"}
+  * Item 4.1 {icon="mif-perm-media"}
+* Item 5 {icon="glyphicon-education"}
+  * Item 5.1 {icon="notebook_with_decorative_cover"}
+  </variable>
+</include>
+
+1. If an item has a specified icon, that icon will be used for it and for subsequent items at that level.
+2. If any item at a level is customized, the first item at that level must also be customized. If not, the list will be invalidated.
+3. You can use any of the icons supported by MarkBind (e.g., Font Awesome, Octicons, Glyphicons and Emojis) as the icon of a list item.
+4. When using icons, omit the `::` around the icon name (e.g., use `{icon="emoji"}` instead of `{icon=":emoji:"}`).
+
+**You can adjust the icon's size by using the `size` attribute.**
+
+<include src="codeAndOutput.md" boilerplate >
+<variable name="highlightStyle">markdown</variable>
+<variable name="code">
+
+* Item 1 {icon="glyphicon-education" size="35px"}
+* Item 2 {icon="glyphicon-education" size="4rem"}
+* Item 3 {icon="glyphicon-education" size="5em"}
+
+</variable>
+</include>
+
+You can utilize any CSS size unit. For more details, refer to the guide on [CSS size units.](https://www.w3schools.com/cssref/css_units.php).
+
+**You can use image as icon.**
+
+<include src="codeAndOutput.md" boilerplate >
+<variable name="highlightStyle">markdown</variable>
+<variable name="code">
+
+* Item 1 {icon="/images/deer.jpg" width="30px" height="17px"}
+* Item 2 {icon="/images/deer.jpg" width="60px" height="34px"}
+* Item 3 {icon="/images/deer.jpg" width="90px" height="51px"}
+* Item 4 {icon="/images/deer.jpg" width="120px" height="68px"}
+* Item 5 {icon="/images/deer.jpg" width="150px" height="84px"}
+
+</variable>
+</include>
+
+The `Width` and `height` is use to specific the dimension for the image icon, similar to size it support all css size units: [CSS Units](https://www.w3schools.com/cssref/css_units.php)
+If the width and height are unspecified, the image's default dimensions will be used.
+
+**The icon's appearance can be further customized by adding a `class` attribute.**
+
+<div id="main-example-markbind">
+
+<include src="codeAndOutput.md" boilerplate >
+<variable name="highlightStyle">markdown</variable>
+<variable name="code">
+* Item 1 <br>
+  Item 1 line 2{icon="/images/deer.jpg" width="60px" height="60px" height="17px" class="rounded"}
+* Item 2
+  Item 2 Continue
+  * Item 2.1 {icon="fas-code-branch" class="text-success"}
+  * Item 2.2 {class="text-danger"}
+  * Item 2.3 
+* Item 3
+  * Item 3.1 
+  * Item 3.2 {class="text-primary"}
+  * Item 3.3 {icon="fas-file-code" class="text-warning"}
```

Comment:
This is feasible. The syntax for such a list would be {text="Step 2", class="text-warning"}. Since we can't reuse the icon attribute, the text will collide with the emoji name.

Reply:
Sound good. As mentioned, we can leave it as a future enhancement.